provider/google_test.go

```diff
@@ -547,18 +609,14 @@ func createZone(t *testing.T, provider *GoogleProvider, zone *dns.ManagedZone) {
 func setupGoogleRecords(t *testing.T, provider *GoogleProvider, endpoints []*endpoint.Endpoint) {
 	clearGoogleRecords(t, provider, "zone-1-ext-dns-test-2-gcp-zalan-do")
 	clearGoogleRecords(t, provider, "zone-2-ext-dns-test-2-gcp-zalan-do")
+	clearGoogleRecords(t, provider, "zone-3-ext-dns-test-2-gcp-zalan-do")
 
 	records, err := provider.Records()
 	require.NoError(t, err)
 
 	validateEndpoints(t, records, []*endpoint.Endpoint{})
 
 	require.NoError(t, provider.CreateRecords(endpoints))
-
-	records, err = provider.Records()
-	require.NoError(t, err)
-
-	validateEndpoints(t, records, endpoints)
```

Comment:
Can we leave this in somehow? The idea is that `setupGoogleRecords` prepares a test case with the given `endpoints` and this validation ensures that this happens correctly. It can be difficult to debug a test case that relies on some existing records if there's an error in `setupGoogleRecords` itself.

Reply:
We could add an additional struct of endpoints that are expected to be filtered out.
( point being the full input into this function might not be the records you expect to get created -- that's why this was removed )

Reply:
I would prefer that. I sent you a possible solution: https://github.com/stealthybox/external-dns/pull/1